### PR TITLE
Add section on how to upgrade Golang in CircleCI

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Add section on how to upgrade Golang in CircleCI

### CODE REVIEW

File: .circleci/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# CircleCI
+
+CircleCI is used to run unit tests on Unix env.
+
+## Upgrading Golang version
+
+/!\ Disclaimer: the datadog-agent-runner-circle should never be used for anything else than CircleCI tests /!\
```
Comment (KSerrania) [inline .circleci/README.md]:
```suggestion
/!\ Disclaimer: the datadog-agent-runner-circle image should never be used for anything else than CircleCI tests /!\
```

File: .circleci/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Change the Golang version in this file `images/runner/Dockerfile#L13`.
```
Comment (KSerrania) [inline .circleci/README.md]: I'd only link the file, not the line (there's no guarantee the go version definition will remain on line 13)

File: .circleci/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+Then locally build and push the new image using
+`datadog/datadog-agent-runner-circle:go<new golang version>` for the images
```
Comment (KSerrania) [inline .circleci/README.md]:
```suggestion
`datadog/datadog-agent-runner-circle:go<new golang version>` for the image's
```

File: .circleci/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+name. You will need write access to that repo on DockerHub (the agent's team
```
Comment (KSerrania) [inline .circleci/README.md]:
```suggestion
name. You will need write access to that repo on DockerHub (the Agent's team
```

File: .circleci/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+should have them).
```
Comment (KSerrania) [inline .circleci/README.md]:
```suggestion
should have it).
```

File: .circleci/README.md (hunk 2 above)
Comment (hush-hush) [reply .circleci/README.md]: I don't know for the image, I could not found where it's used

**Pull request merged**